src/config/env.c

```diff
@@ -147,15 +147,32 @@ void printFTLenv(void)
 
 static struct env_item *__attribute__((pure)) getFTLenv(const char *key)
 {
+	// Normalize the environment variable to conventional names,
+	// i.e. uppercase letters, digits, and the <underscore> ( '_' )
+	// Exact case matches will still work.
+	// See: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html
+	char *env_key = strdup(key);
+	int env_length = strlen(env_key);
+	for (int i = 0; i < env_length; i++)
+	{
+		if (env_key[i] == '.')
+			env_key[i] = '_'; // keys are pre-processed, but just in case
+		else
+			env_key[i] = toupper(env_key[i]);
+	}
 	// Iterate over all known FTLCONF environment variables
 	for(struct env_item *item = env_list; item != NULL; item = item->next)
 	{
 		// Check if this is the requested key
-		if(strcmp(item->key, key) == 0)
+		if(strcmp(item->key, key) == 0 || strcmp(item->key, env_key) == 0)
```

Comment:
Why don't we simply use the case insensitive variant of `strcmp()` here instead? With this we won't need to create the uppercase variants of the keys.
```suggestion
		if(strcasecmp(item->key, key) == 0)
```
and then all the remaining code you added to this file can be removed.

Reply:
That makes much cleaner/nicer code, sorry I didn't submit that originally.

And it won't be shown anything that doesn't start with FTLCONF_ (that gets checked before this method), but other than that the documentation won't mention the flexibility.

Reply:
I wonder why you marked this conversation as resolved, do you still plan submitting the simplified code I suggested?

Reply:
Of course that's confusing, I made the changes and pushed them to the wrong branch!